services/QuillLMS/app/controllers/snapshots_controller.rb

```diff
@@ -134,10 +134,18 @@ def options
 
   private def retrieve_cache_or_enqueue_worker(worker)
 
-    previous_start, previous_end, current_start, current_end = Snapshots::Timeframes.calculate_timeframes(snapshot_params[:timeframe],
+    previous_start, previous_end, timeframe_start, timeframe_end = Snapshots::Timeframes.calculate_timeframes(snapshot_params[:timeframe],
```

Comment:
Would this be cleaner if you passed `previous_timeframe` to the `calculate_timeframes` method and returned the correct start/end from there, versus swapping below. This current setup does 4 calculations for timeframes, then always throws away 2.

Looking at the calculate_timeframe code, you could use the `previous_timeframe` boolean to determine whether to run/return `[:previous_start, :previous_end]` or `[:current_start, :current_end] `